src/app/containers/DashboardView/DashboardView.tsx

```diff
@@ -31,18 +34,15 @@ class DashboardView extends React.Component<IProps, IState> {
   }
 
   componentWillReceiveProps(nextProps: IProps) {
-    if (nextProps.hasLoadedState && !this.props.hasLoadedState) {
-      const pathname = this.props.location.pathname
-      // todo: refactor - store selected portfolio id in the state maybe?
-      if (pathname.split('/').length === 2) {
-        const firstPortfolio = _.first(this.props.userStore!.portfolios)
-        if (firstPortfolio) {
-          this.props.history.push(`${this.props.match.url}/${firstPortfolio.id}`)
-        }
-      }
+    if (nextProps.activePortfolioId !== this.props.activePortfolioId) {
+      this.navigateToPortfolio(nextProps.activePortfolioId)
     }
   }
 
+  private navigateToPortfolio(id: string) {
+    this.props.history.push(`${this.props.match.url}/${id}`)
```

Comment:
hm, tai jei jau visas tas state nukeliauja is url i store tai gal reiketu ir nebenaudoti tiesiogiai to history push?